src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,581 @@
+      case "Validate_EventHub" => "Consuming data from EventHub."
+      case "Valid_Excluded_Scopes" => "Excluded scope can be audit:sparkEvents:jobs:clusters:clusterEvents:notebooks:pools:accounts."
+      case "Storage_Access" => "ETL_STORAGE_PREFIX should have read,write and create access"
+    }
+  }
+
+  /**
+   * Performs mandatory validation before each deployment.
+   */
+  private[overwatch] def performMandatoryValidation = {
+    logger.log(Level.INFO, "Performing mandatory validation")
+    validateFileExistance
+    makeDataFrame
+    val groupedRuleSet = RuleSet(inputDataFrame, by = "deploymentId")
+    groupedRuleSet.add(validateEtlStorage)
+    validateRuleAndUpdateStatus(groupedRuleSet, true)
+    if (validationStatus.toDS().filter("validated==false").count > 0) { //Checks for failed validations
+      throw new BadConfigException(getSimpleMsg("Common_ETLStoragePrefix"))
+    }
+    storagePrefixAccessValidation(inputDataFrame.head(), true)
+  }
+
+  /**
+   * Entry point of the validation.
+   *
+   * @return
+   */
+  private[overwatch] def performValidation: Dataset[DeploymentValidationReport] = {
+    //Primary validation //
+    validateFileExistance
+    makeDataFrame
+    println("Parallelism :" + parallelism)
+
+    //csv data validation
+    val processingStartTime = System.currentTimeMillis();
+    val groupedRuleSet = RuleSet(inputDataFrame, by = "deploymentId")
+    groupedRuleSet.add(validateEtlStorage)
+    groupedRuleSet.add(validateETLDatabaseName)
+    groupedRuleSet.add(validateConsumerDatabaseName)
+    validateRuleAndUpdateStatus(groupedRuleSet, true)
+
+    val nonGroupedRuleSet = RuleSet(inputDataFrame)
+    nonGroupedRuleSet.add(validateCloud)
+    nonGroupedRuleSet.add(validateAPIURL)
+    nonGroupedRuleSet.add(validateSecretScope)
+    nonGroupedRuleSet.add(validateSecreteKeyDBPAT)
+    nonGroupedRuleSet.add(validatePrimordialDate)
+    nonGroupedRuleSet.add(validateMaxDays)
+    nonGroupedRuleSet.add(validateOWScope())
+    validateRuleAndUpdateStatus(nonGroupedRuleSet, false)
+    val processingEndTime = System.currentTimeMillis();
+    println("Rule engine validation Duration in sec :" + (processingEndTime - processingStartTime) / 1000)
+
+    //Connectivity validation
+    val inputRow = inputDataFrame.collect().par
+    val taskSupport = new ForkJoinTaskSupport(new ForkJoinPool(parallelism))
+    inputRow.tasksupport = taskSupport
+    inputRow.map(validateApiUrl)
+    val processingEtURL = System.currentTimeMillis();
+    println("URL validation Duration in sec :" + (processingEtURL - processingEndTime) / 1000)
+
+    //Cloud specific validation s3/EH
+    inputRow.map(cloudSpecificValidation)
+    val processingCould = System.currentTimeMillis();
+    println("Cloud specific validation Duration in sec :" + (processingCould - processingEtURL) / 1000)
+    storagePrefixAccessValidation(inputRow.head)
+
+    inputDataFrame.unpersist()
+    validationStatus.toDS()
```

Comment:
strongly type this
validationStatus.as[MyObjectType] -- this often results in encoder issues.

Reply:
done